packages/event-sourcing-firestore/src/nested/firestore.event-store.ts

```diff
@@ -4,66 +4,117 @@ import {
   IChange,
   IFact,
 } from "@ddd-ts/core";
-import type { FirestoreTransaction } from "@ddd-ts/store-firestore";
+import {
+  DefaultConverter,
+  FirestoreTransaction,
+} from "@ddd-ts/store-firestore";
 import * as fb from "firebase-admin";
 
 export class NestedFirestoreEventStore {
-  constructor(public readonly firestore: fb.firestore.Firestore) {}
+  constructor(
+    public readonly firestore: fb.firestore.Firestore,
+    public readonly converter: fb.firestore.FirestoreDataConverter<fb.firestore.DocumentData> = new DefaultConverter(),
+  ) {}
 
   get collection() {
     return this.firestore.collection("event-store");
   }
 
   async close() {}
 
-  async append(
+  async bulkAppend(
+    toAppend: {
+      streamId: AggregateStreamId;
+      changes: IChange[];
+      expectedRevision: number;
+    }[],
+    trx: FirestoreTransaction,
+  ) {
+    await Promise.all(
+      toAppend.map(
+        async ({ streamId, expectedRevision }) =>
+          await this.lockDocument(streamId, expectedRevision, trx),
+      ),
+    );
+
+    await Promise.all(
+      toAppend.map(async ({ streamId, changes, expectedRevision }) => {
+        await this.commitChanges(streamId, changes, expectedRevision, trx);
+      }),
+    );
+  }
```

Comment:
you could literrally use the bulk service by firestore

Reply:
Bulk service is not transactionnal i think